connman/src/network.c

```diff
@@ -1538,6 +1549,7 @@ int __connman_network_disconnect(struct connman_network *network)
 		return -EUNATCH;
 
 	network->connecting = false;
+	network->disconnecting = true;
```

Comment:
Before anyone comments I totally agree that network.c would start to benefit a network state machine...